nengo/processes.py

```diff
@@ -264,6 +278,40 @@ def step(t):
         return step
 
 
+class PresentInput(Process):
+    """Present a series of inputs, each for the same fixed length of time.
+
+    Parameters
+    ----------
+    inputs : array_like
+        Inputs to present, where each row is an input. Rows will be flattened.
+    presentation_time : float
+        Show each input for `presentation_time` seconds.
+    """
+    inputs = NdarrayParam(shape=('...',))
+    presentation_time = NumberParam(low=0, low_open=True)
+
+    def __init__(self, inputs, presentation_time):
+        self.inputs = inputs
+        self.presentation_time = presentation_time
+        super(PresentInput, self).__init__(
+            default_size_out=self.inputs[0].size)
+
+    def make_step(self, size_in, size_out, dt, rng):
+        assert size_in == 0
+        assert size_out == self.inputs[0].size
+
+        n = len(self.inputs)
+        inputs = self.inputs.reshape(n, -1)
+        presentation_time = float(self.presentation_time)
+
+        def step_image_input(t):
```

Comment:
The names of these functions don't really matter, though they might appear in debugging sometimes. I'm wondering if we should always have them be `step_<class name>`, so in this case `step_presentinput`.